### PR TITLE
AttributePartitioningStrategy config protocol changes

### PR DESCRIPTION
**Pull request by ivanthescientist**
Adds custom type for `PartitioningStrategyConfig` entities as well as field for transferring list of partitioning attribute configs.

### CODE REVIEW

**Review by viliam-durina — DISMISSED** [review]

File: protocol-definitions/DynamicConfig.yaml [hunk 1]
```diff
@@ -903,6 +903,12 @@ methods:
           since: 2.5
           doc: |
             Tiered-Store configuration
+        - name: partitioningAttributeConfigs
+          type: List_PartitioningAttributeConfig
+          nullable: true
```
Comment (mdumandag) [inline protocol-definitions/DynamicConfig.yaml:908]: just to double check, as most of the other configs here are not nullable: should we make this nullable?

File: protocol-definitions/custom/Custom.yaml [hunk 2]
```diff
@@ -1144,3 +1144,10 @@ customTypes:
         type: boolean
         nullable: false
         since: 2.6
+  - name: PartitioningAttributeConfig
+    since: 2.6
+    params:
+      - name: attributeName
+        type: String
+        nullable: false
+        since: 2.6
```
Comment (mdumandag) [inline protocol-definitions/custom/Custom.yaml:1153]: minor: newline at the end of the file

File: binary/__init__.py [hunk 3]
```diff
@@ -95,6 +95,7 @@
     'MemoryTierConfig',
     'DiskTierConfig',
     'TieredStoreConfig',
+    "PartitioningAttributeConfig",
```
Comment (mdumandag) [inline binary/__init__.py:98]: Minor: I believe this is not needed for now. Maybe we can remove references to non-list field from this PR

File: protocol-definitions/DynamicConfig.yaml [hunk 4]
```diff
@@ -903,6 +903,12 @@ methods:
...
+          since: 2.6
+          doc: |
+            List of Attributes used for creating AttributePartitioningStrategy.
```
Comment (mdumandag) [inline protocol-definitions/DynamicConfig.yaml:911]:
```suggestion
            List of attributes used for creating AttributePartitioningStrategy.
```

File: protocol-definitions/DynamicConfig.yaml (hunk 1 above)
Comment (ivanthescientist) [reply protocol-definitions/DynamicConfig.yaml:908]: I'm not sure to be honest, IndexConfigs are optional and its a list too.

**Review by srknzl — APPROVED** [review]

**Pull request merged**